src/raftstore/store/fsm/store.rs

```diff
@@ -471,7 +471,9 @@ impl<T: Transport, C: PdClient> RaftPoller<T, C> {
                     .schedule_task(prop.region_id, ApplyTask::Proposal(prop));
             }
         }
-        if self.poll_ctx.need_flush_trans {
+        if self.poll_ctx.need_flush_trans
+            && (!self.poll_ctx.kv_wb.is_empty() || !self.poll_ctx.raft_wb.is_empty())
+        {
             self.poll_ctx.trans.flush();
```

Comment:
Seems messages are flushed before disk write. If the messsages contain vote messages, could the peer vote twice in a term if the machine fail after here and before disk write?

Reply:
If there are pending messages, it means they are from leader, it's OK to flush before writing disk. Also related to pingcap/raft-rs#292.